File: Easygym.Domain/Entities/Workout.cs
using System.ComponentModel.DataAnnotations;

namespace Easygym.Domain.Entities
{
    public class Workout
    {
        public int Id { get; set; }
        [StringLength(50)]
        public string? Name { get; set; }
        [StringLength(50)]
        public string? Description { get; set; }
        public required int TraineeId { get; set; }
        public User? Trainee { get; set; }
        public required List<Set> Sets { get; set; }
        public int? RestTimeSeconds { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.Now;
    }
}

File: Client/src/types/Workout.ts
import { User } from '@/types/User';

export interface Set {
  id: number;
  name: string;
  description?: string;
  repetitions: number;
  weight?: number;
}

export interface Workout {
  id: number;
  name?: string;
  description?: string;
  traineeId: number;
  trainee?: User;
  sets: Set[];
  restTimeSeconds: number;
  createdAt: string;
}

export interface CreateWorkoutRequest {
  name?: string;
  description?: string;
  traineeId: number;
  sets: Omit<Set, 'id'>[];
  restTimeSeconds?: number;
}

export interface UpdateWorkoutRequest {
  name?: string;
  description?: string;
  sets?: Set[] | Omit<Set, 'id'>[];
  restTimeSeconds?: number;
}